clang/docs/analyzer/checkers.rst

```diff
@@ -129,6 +129,8 @@ The ``SuppressAddressSpaces`` option suppresses
 warnings for null dereferences of all pointers with address spaces. You can
 disable this behavior with the option
 ``-analyzer-config core.NullDereference:SuppressAddressSpaces=false``.
+Value of this option is used for checker
```

Comment:
```suggestion
Value of this option is also used for checker
```

It's really weird to see that the option of a different checker influences some other checker.
That that point maybe the checker option should be hoisted into an analyzer config option instead.
WDYT?

Reply:
I'd prefer adding and the analyzer option in this PR to prevent a weird half-ways state.
We should also announce this new checker in the release notes along with the change that the option `SuppressAddressSpaces` was hoisted to bring visibility.